tests/test_flow/test_reload.py

```diff
@@ -40,216 +40,240 @@ def _write_module(name, contents):
     sys.dont_write_bytecode = dont_write_bytecode
 
 
+@pytest.fixture(params=["reload", "reloading"])
+def reload_method(request):
+    return request.param
+
+
 @pytest.fixture
-def write_flow_module(write_module):
-    def _write_flow_module(x):
-        write_module(
-            "flow_module",
-            f"""
-                import bionic as bn
+def flow_tester(reload_method, tmp_path, write_module):
+    class FlowTester:
+        """
+        A helper class for testing dynamically written flows.
 
-                builder = bn.FlowBuilder('flow')
-                builder.assign('x', {x})
+        If the reload method is reloading(), whenever the method is called, we get a
+        new copy of the flow with reloaded modules. We need to verify that the
+        entities of the original flow instance give the same values when computed as
+        before the reload.
+        """
 
-                @builder
-                def x_doubled(x):
-                    return x*2
+        def __init__(self):
+            self.flow = None
+            self.expected_results_by_flow = {}
 
-                flow = builder.build()
-            """,
-        )
+        def write_tester_flow(self, contents):
+            write_module(
+                "flow_module",
+                f"""
```

Comment:
You can avoid this awkward indent by breaking this flow string into three dedented strings (before `contents`, `contents`, and after `contents`) and joining them.

Reply:
Or you can do what Janek suggested.

Reply:
I like this; it maintains the test's simplicity.